proxy-node-translator/src/test/java/uk/gov/ida/notification/translator/apprule/HubResponseTranslatorAppRuleTests.java

```diff
@@ -91,78 +97,96 @@ public void setup() throws Throwable {
         KeyPairConfiguration signingKeyPair = translatorAppRule.getConfiguration().getConnectorFacingSigningKeyPair();
         SignatureValidator signatureValidator = new CredentialFactorySignatureValidator(new SigningCredentialFactory(entityId -> singletonList(signingKeyPair.getPublicKey().getPublicKey())));
 
-        Response eidasResponse = extractEidasResponse(buildSignedHubResponse());
+        Response eidasResponse = extractEidasResponseFromTranslator(buildSignedHubResponse());
 
         Signature signature = eidasResponse.getSignature();
 
-        assertNotNull("SAML Response needs to be signed", signature);
+        assertNotNull("EIDAS SAML Response needs to be signed", signature);
         assertTrue("Invalid signature", signatureValidator.validate(eidasResponse, null, Response.DEFAULT_ELEMENT_NAME));
         assertEquals(SignatureConstants.ALGO_ID_SIGNATURE_RSA_SHA256, signature.getSignatureAlgorithm());
     }
 
     @Test
     public void shouldReturnAnEncryptedEidasResponse() throws Exception {
-        Response eidasResponse = extractEidasResponse(buildSignedHubResponse());
+        Response eidasResponse = extractEidasResponseFromTranslator(buildSignedHubResponse());
         assertEquals(1, eidasResponse.getEncryptedAssertions().size());
         assertThat(eidasResponse.getAssertions()).isEmpty();
     }
 
     @Test
-    @Ignore
-    public void postingHubResponseShouldReturnEidasResponseForm() throws Exception {
-        Response hubResponse = buildSignedHubResponse();
-        Credential decryptingCredential = new TestCredentialFactory(TEST_PUBLIC_CERT, TEST_PRIVATE_KEY).getDecryptingCredential();
+    public void shouldDecryptAndReadEidasAssertion() throws Exception {
 
-        Response eidasResponse = extractEidasResponse(hubResponse);
-        Response decryptedEidasResponse = decryptResponse(eidasResponse, decryptingCredential);
+        Response hubResponse = buildSignedHubResponse();
+        Response eidasResponse = extractEidasResponseFromTranslator(hubResponse);
+        Response decryptedEidasResponse = decryptResponse(eidasResponse, eidasDecryptingCredential);
         Assertion eidasAssertion = decryptedEidasResponse.getAssertions().get(0);
-        Element attributeStatement = MARSHALLER.marshallToElement(eidasAssertion.getAttributeStatements().get(0));
 
         assertEquals(hubResponse.getInResponseTo(), eidasResponse.getInResponseTo());
+        assertEquals(2, eidasAssertion.getStatements().size());
         assertEquals(1, eidasAssertion.getAttributeStatements().size());
         assertEquals(1, eidasAssertion.getAuthnStatements().size());
-
-        assertEquals("Jazzy Harold", attributeStatement.getFirstChild().getTextContent());
     }
 
     @Test
-    @Ignore
-    public void shouldNotAcceptUnsignedHubResponse() throws Exception {
-        javax.ws.rs.core.Response response = postHubResponseToTranslator(buildUnsignedHubResponse());
```

Comment:
`buildUnsignedHubResponse` is now unused - can we remove it as well

Reply:
These are deleted lines :)

Reply:
Yes, but the deleted code used to call the method `buildUnsignedHubResponse` - after deletion that method is no longer used. Needs to be deleted as well

Reply:
OK deleted.